src/ngx_http_srcache_headers.c

```diff
@@ -104,6 +108,126 @@ ngx_http_srcache_header_t  ngx_http_srcache_headers_in[] = {
 
 
 ngx_int_t
+ngx_http_srcache_process_expire(ngx_http_request_t *r, ngx_table_elt_t *h,
+    ngx_uint_t offset)
+{
+    ngx_table_elt_t  *ho, **ph;
+
+    ngx_uint_t temp_time;
+
+    ngx_http_srcache_ctx_t *ctx;
+    ngx_http_srcache_ctx_t *pr_ctx;
+
+    ctx = ngx_http_get_module_ctx(r, ngx_http_srcache_filter_module);
+
+    if (ctx == NULL) {
+      return NGX_ERROR;
+    }
+
+    if (h != NULL){
+      temp_time = ngx_http_parse_time (h->value.data, h->value.len);
+
+      if (temp_time == NGX_ERROR){
+        return NGX_ERROR;
+      }
+
+    	int first_request;
+
+    	if (r->parent != NULL) {
+
+    	  pr_ctx = ngx_http_get_module_ctx(r->parent,
+    					   ngx_http_srcache_filter_module);
+
+    	  if (pr_ctx == NULL) {
+    	    return NGX_ERROR;
+    	  }
+    	  first_request = pr_ctx->first_request;
+    	}	else {
+        first_request = ctx->first_request;
+      }
+
+    	if (first_request){  
+    	  ctx->first_request = 0;
+
+    	  if (r->parent != NULL) {
+    	    pr_ctx->first_request = 0;
+        }
+    	  
+    	  if (temp_time <= ngx_time()){
```

Comment:
Style: need a space before `{`. Please fix your coding style to be consistent with the rest of the code base. There are just too many inconsistencies. Alas.